### ISSUE TITLE
UDP listening and can respond, but won't

### ISSUE THREAD

**Issue by Tzrlk** [lloesche/valheim-server-docker#479]
I've been banging my head against this for days.
* Server running in ECS
* I've got 2456-2458 open and forwarded
* HTTP access to all tcp endpoints work fine
* Status returns no errors.
* No rejected udp traffic in network flow-logs
* Can successfully netcat from local to server on 2458 (the only port not being used by the server)
* Nothing obvious in the supervisor or steam logs that I can find.
* Not present in either steam server, nor in-game lists
* Direct connections to ip fail to connect on any port

I've pretty much eliminated networking, with the only thing left being to completely disable IPv6.
Started looking into log aggregation via fluentd, etc. since it looks like some log info doesn't make it's way to stdout/err.

Any ideas where I should be looking to figure this out? Errors that don't actually look like errors?
This is so unbelievably frustrating to debug through.

**Comment by Tzrlk** [comment]
Disabling ipv6 in sysctl didn't help either.

**Comment by Tzrlk** [comment]
At the moment, I'm giving it 2vcpus and 6G of ram. Should be way more than enough to handle the load. I haven't seen it restarting either.

I'll have to check it again, but I'm pretty sure the steam connection logs have no record of my connection attempts. My next stack improvement plan is to aggregate the logs using fluentd or something so I can scan them in one go.

All the configs are in my repo (tzrlk/valheim-server) under src/docker and src/terraform. Ignore the CDK stuff for now.

**Comment by dfibuch** [comment]
> At the moment, I'm giving it 2vcpus and 6G of ram. Should be way more than enough to handle the load. I haven't seen it restarting either.
> 
> I'll have to check it again, but I'm pretty sure the steam connection logs have no record of my connection attempts. My next stack improvement plan is to aggregate the logs using fluentd or something so I can scan them in one go.
> 
> All the configs are in my repo (tzrlk/valheim-server) under src/docker and src/terraform. Ignore the CDK stuff for now.

Do you mean this repo https://github.com/Tzrlk/iac-valheim? I'm by no means an expert as I only set my first server up last week but I can have a look. Also, I'm hosting mine in Azure so cant comment on the Amazon hosting bit, not sure if you got the ability to try your docker-compose in Azure? See if its environmental.

**Comment by Tzrlk** [comment]
Yup, that's the one. Though it'd take a rewrite of the whole Terraform stack to get it up in Azure. Maybe after I've converted it to CDK (which is what I do when I'm stuck). Gonna see if netcat picks up steam/valheim traffic and/or stick an nginx sidecar in there to monitor the connections.

**Comment by dfibuch** [comment]
Ok so one thing I noticed is you haven't exposed port 2458/udp. 

Also, are you getting all your files getting created correctly on your volume? The `volumes` config you have is quite different, personally I don't quite get what the `/config` and `/opt/valheim` bit does but that's how I have it set (and the default example also).

```
    volumes: 
      - configfiles:/config
      - datafiles:/opt/valheim
```

Both `configfiles` and `datafiles` point to the root of the same physical volume.

When I was having my issues it looked like everything was running and working but I kept having the same error and the server kept stopping. Check your logs for "Shutdown complete" to see if its actually shutting itself off.

**Comment by Tzrlk** [comment]
I'll need to update my docker compose for local testing, but the actual config is the ECS container definition in src/terraform/ecs_service.tf. It exposes 2456-2458/udp, and doesn't mount any volumes. I'm using s3 sync to backup and restore the world state.

Funnily enough, when the server is running, it doesn't even bother listening on 2458, just 6 and 7. Since I'd put the network rules in there anyway, I confirmed the port was working fine with netcat using it.

I'll update my docker compose and see if I can replicate the issue locally.
